### PR TITLE
ALS-4465: Add endpoint to support auto completer for genes

### PR DESCRIPTION
**Pull request by ramari16**
* Adds endpoint to support autocompleter for genes (and any categorical genomic variable)
* Some refactoring to support integration testing. Related PR incoming

### CODE REVIEW

File: service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java [hunk 1]
```diff
@@ -283,6 +284,31 @@ public Response querySync(QueryRequest resultRequest) {
...
+			throw new IllegalArgumentException("Page must be greater than 0");
+		}
+		if (size < 1) {
+			throw new IllegalArgumentException("Size must be greater than 0");
+		}
+		final List<String> matchingValues = abstractProcessor.searchInfoConceptValues(conceptPath, query);
+		return new PaginatedSearchResult<>(getPage(matchingValues, page, size), page, matchingValues.size());
+	}
+
+	public static <T> List<T> getPage(List<T> list, int page, int size) {
```
Comment (ramari16) [inline service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java:306]: I should probably move this to a util class or somehing

File: etl/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/etl/phenotype/CSVLoader.java [hunk 2]
```diff
@@ -31,15 +31,23 @@ public class CSVLoader {
 
 	private static final int DATETIME = 4;
 
+	private static String HPDS_DIRECTORY = "/opt/local/hpds/";
+
 	public static void main(String[] args) throws IOException {
-		store.allObservationsStore = new RandomAccessFile("/opt/local/hpds/allObservationsStore.javabin", "rw");
+		if (args.length > 0) {
+			HPDS_DIRECTORY = args[0];
+			if (!HPDS_DIRECTORY.endsWith("/")) {
```
Comment (Luke-Sikina) [inline etl/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/etl/phenotype/CSVLoader.java:39]: You don't actually need this check! Just append the slash - `my/path//` and `my/path/` are the same in UNIX.

File: processing/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/processing/AbstractProcessor.java [hunk 3]
```diff
@@ -526,6 +535,19 @@ protected BigInteger createMaskForPatientSet(Set<Integer> patientSubset) {
 	public FileBackedByteIndexedInfoStore getInfoStore(String column) {
 		return infoStores.get(column);
 	}
+
+	public List<String> searchInfoConceptValues(String conceptPath, String query) {
+		final FileBackedByteIndexedInfoStore store = getInfoStore(conceptPath);
+		if (store == null) {
+			throw new IllegalArgumentException("Concept path: " + conceptPath + " not found");
+		}
+		final List<String> allValues = new ArrayList<>(store.getAllValues().keys());
```
Comment (Luke-Sikina) [inline processing/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/processing/AbstractProcessor.java:544]: Are you actually observing a measurable performance from making these explicitly final, or is this just for the labdas? If it's the latter, you don't need to be explicit - the compiler will mark them as effectively final and be happy.

File: processing/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/processing/AbstractProcessor.java [hunk 4]
```diff
@@ -526,6 +535,19 @@ protected BigInteger createMaskForPatientSet(Set<Integer> patientSubset) {
...
+		// todo: add cache for sorted values if performance is an issue
+		return allValues.stream()
```
Comment (Luke-Sikina) [inline processing/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/processing/AbstractProcessor.java:546]:
```suggestion
		// todo: add cache for sorted values if performance is an issue
		return store.getAllValues().keys().stream()
```

File: service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java (hunk 1 above)
Comment (Luke-Sikina) [reply service/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/service/PicSureService.java:306]: Yea. Ideally, I'd move to a util class, make it not static, inject it, and cover it with some unit tests that demonstrate the edge cases.

File: processing/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/processing/AbstractProcessor.java [hunk 5]
```diff
@@ -526,6 +536,35 @@ protected BigInteger createMaskForPatientSet(Set<Integer> patientSubset) {
...
+					.filter(variableValue -> variableValue.toUpperCase().contains(query.toUpperCase()))
+					.collect(Collectors.toList());
+		} catch (UncheckedExecutionException e) {
+			if(e.getCause() instanceof RuntimeException) {
+				throw (RuntimeException) e.getCause();
+			}
+			throw e;
+		}
+	}
+	private final LoadingCache<String, List<String>> infoStoreValuesCache = CacheBuilder.newBuilder().build(new CacheLoader<>() {
```
Comment (ramari16) [inline processing/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/processing/AbstractProcessor.java:552]: Move to the top/constructor
